### PR TITLE
Go extractor - Add Go dependencies tree command

### PR DESCRIPTION
**Pull request by barbelity**

### CODE REVIEW

File: build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoPublish.java [hunk 1]
```diff
@@ -54,6 +53,8 @@
      */
     public GoPublish(ArtifactoryBuildInfoClientBuilder clientBuilder, ArrayListMultimap<String, String> properties, String repo, Path path, String version, Log logger) throws  IOException {
         super(clientBuilder, path, logger);
+        this.goDriver = new GoDriver(GO_CLIENT_CMD, null, path.toFile(), logger);
+        this.moduleName = goDriver.getModuleName();
```
Comment (yahavi) [inline build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoPublish.java]: These 2 lines repeat in GoRun. Please move them up to GoCommand.

File: build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/TestUtils.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+public class TestUtils {
+
+    public static final Path PROJECTS_ROOT = Paths.get(".").toAbsolutePath().normalize().resolve(Paths.get("src", "test", "resources", "org", "jfrog", "build", "extractor"));
+
+    public static Path createProjectDir(String targetDir, File projectOrigin) throws IOException {
+        File projectDir = Files.createTempDirectory(targetDir).toFile();
+        FileUtils.copyDirectory(projectOrigin, projectDir);
+        return projectDir.toPath();
+    }
+
```
Comment (yahavi) [inline build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/TestUtils.java]:
```suggestion
```

File: build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/dependencyTree/GoDependencyTreeTest.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+import static org.testng.Assert.*;
+
+/**
+ * Created by Bar Belity on 16/02/2020.
+ */
+public class GoDependencyTreeTest {
+
+    protected static final Log log = new TestingLog();
+
```
Comment (yahavi) [inline build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/dependencyTree/GoDependencyTreeTest.java]:
```suggestion
```

File: build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/dependencyTree/GoDependencyTreeTest.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+            Path projectDir = TestUtils.createProjectDir("treeProject2", TestUtils.PROJECTS_ROOT.resolve("treeProject2").toFile());
+            GoDriver goDriver = new GoDriver("", null, projectDir.toFile(), log);
+            DependenciesTree dt = GoDependencyTree.createDependenciesTree(goDriver);
+            validateDependenciesTreeResults(expected, dt);
+        } catch (IOException ex) {
+            fail(ExceptionUtils.getStackTrace(ex));
+        }
+    }
+
+    private Map<String, List<String>> getAllDependeciesForTest() {
```
Comment (yahavi) [inline build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/dependencyTree/GoDependencyTreeTest.java]:
```suggestion
    private Map<String, List<String>> getAllDependenciesForTest() {
```

File: build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/dependencyTree/GoDependencyTreeTest.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        return new HashMap<String, List<String>>(){
+            {
+                put("my/pkg/name1", new ArrayList<>(Arrays.asList(
```
Comment (yahavi) [inline build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/dependencyTree/GoDependencyTreeTest.java]: We don't need new ArrayList if the list is immutable. Is suggest this:

```java
put("my/pkg/name1", Arrays.asList(
        "github.com/jfrog/directDep1@v0.1",
        "github.com/jfrog/directDep2@v0.2",
        "github.com/jfrog/directDep3@v0.3"));
put("github.com/jfrog/directDep1@v0.1", Arrays.asList(
        "github.com/jfrog/indirectDep1-1@v1.1",
        "github.com/jfrog/indirectDep2-1@v1.2"));
put("github.com/jfrog/directDep2@v0.2", Collections.singletonList(
        "github.com/jfrog/indirectDep1-2@v2.1"));
put("github.com/jfrog/indirectDep1-1@v1.1", Collections.singletonList(
        "github.com/jfrog/indirectIndirectDep1-1-1@v1.1.1"));
```

File: build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoPublish.java (hunk 1 above)
Comment (barbelity) [reply build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoPublish.java]: I don't see the benefit of doing that.
In each case it happens at different stage,  and moving it to a dedicated method seems more confusing than helping.

Comment (yahavi) [reply build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoPublish.java]: Can we do both in the constructor of GoCommand?

**Pull request merged**